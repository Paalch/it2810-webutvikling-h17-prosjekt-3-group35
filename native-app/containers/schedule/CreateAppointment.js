import React from 'react';
import { StyleSheet, Text, View, TextInput, Form, Button, DatePickerAndroid, TimePickerAndroid, Modal } from 'react-native';
import moment from 'moment'

export default class CreateAppointment extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            date: moment(),
            time: moment().set('minute', 0),
            details: '',
            label: '',
            idCounter: 0,
            appointments: this.props.appointments,
            color: '#337ab7'
        };
        this.handleDateChange = this.handleDateChange.bind(this);
        this.handleTimeChange = this.handleTimeChange.bind(this);
        this.createAppointment = this.createAppointment.bind(this);
        this.updateDetails = this.updateDetails.bind(this);
        this.setColor = this.setColor.bind(this);
        this.disabledMinutes = this.disabledMinutes.bind(this);
    }

    handleDateChange(date) {
        this.setState({
            date: date
        });
    }

    handleTimeChange(time) {
        this.setState({
            time: time
        });
    }

    updateDetails(details) {
        this.setState({
            details: details.target.value
        });
    }

    setColor(c) {
        this.setState({color: c});
    }

    disabledMinutes() {
        let minutes = [];
        for (let i = 1; i <= 60; i++) {
            if (i % 15 !== 0) {
                minutes.push(i);
            }
        }
        return minutes;
    }
    

    createAppointment() {
        let appointments = this.state.appointments;

        let id = this.state.idCounter;
        let color = this.state.color;

        appointments.push([
            this.state.date,
            this.state.time,
            this.state.details,
            id,
            color]);

        this.setState({
            appointments: appointments,
            idCounter: this.state.idCounter + 1
        });
        this.props.onCreateAppointment(this.state.appointments);
        this.setState({details: '', color: '#337ab7', date: moment(), time: moment().set('minute', 0)});
        this.props.closeNewAppointment();
    };

    render() {
        return (
                <Modal onRequestClose={this.props.closeNewAppointment} onShow={this.props.showModal}>
                    <Modal.Header closeButton>
                        <Modal.Title>Create new event</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        <View className={'modal-row'}>
                            <View>
                                <Text>Date:</Text>
                                <DatePickerAndroid
                                    selected={this.state.date}
                                    onChange={this.handleDateChange}
                                    locale={"no-bm"}
                                />
                            </View>
                            <View>
                                <Text>Time:</Text>
                                <TimePickerAndroid
                                    showSecond={false}
                                    defaultValue={moment().set('minute', 0)}
                                    onChange={this.handleTimeChange}
                                    disabledMinutes={this.disabledMinutes}
                                    hideDisabledOptions={true}

                                />
                            </View>
                        </View>
                        <View className={'modal-row'}>
                            <View>
                                <Text>Label:</Text>
                            </View>
                            <Form>
                                <View>
                                    <TextInput
                                        id={'label1'}
                                        type={'radio'}
                                        name={'radioGroup'}
                                        className={'modal-label-picker'}
                                        onChange={() => this.setColor('#337ab7')}/>
                                </View>
                                <View>
                                    <TextInput
                                        id={'label2'}
                                        type={'radio'}
                                        name={'radioGroup'}
                                        className={'modal-label-picker'}
                                        onChange={() => this.setColor('#A25188')}/>
                                </View>
                                <View>
                                    <TextInput
                                        id={'label3'}
                                        type={'radio'}
                                        name={'radioGroup'}
                                        className={'modal-label-picker'}
                                        onChange={() => this.setColor('#447884')}/>
                                </View>
                                <View>
                                    <TextInput
                                        id={'label4'}
                                        type={'radio'}
                                        name={'radioGroup'}
                                        className={'modal-label-picker'}
                                        onChange={() => this.setColor('#CA6573')}/>
                                </View>
                                <View>
                                    <TextInput
                                        id={'label5'}
                                        type={'radio'}
                                        name={'radioGroup'}
                                        className={'modal-label-picker'}
                                        onChange={() => this.setColor('#7B498D')}/>
                                </View>
                            </Form>
                        </View>
                        <View className={'modal-row'}>
                            <View>
                                <Text>Schedule description:</Text>
                            </View>
                            <View>
                                <TextInput
                                    type={'text'}
                                    placeholder={''}
                                    value={this.state.details}
                                    onChangeText={this.updateDetails}
                                />
                            </View>
                            <View>
                                <Button
                                    onPress={this.createAppointment}
                                    bsStyle={'primary'}
                                    className={'modal-schedule-button'}
                                    title={"Schedule event"}/>
                            </View>
                        </View>
                    </Modal.Body>
                </Modal>
        );
    }
}

const styles = StyleSheet.create({
    container: {

    },
});
