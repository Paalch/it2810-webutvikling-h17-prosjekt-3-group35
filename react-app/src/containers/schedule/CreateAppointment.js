import React, {Component} from 'react'
import {Col, Row, FormControl, Button, Modal} from 'react-bootstrap'
import moment from 'moment'
import DatePicker from 'react-datepicker'
import TimePicker from 'rc-time-picker';

import 'rc-time-picker/assets/index.css';
import 'react-datepicker/dist/react-datepicker.css';


class CreateAppointment extends Component {

  constructor(props) {
    super(props);
    this.state = {
      date: moment(),
      time: moment(),
      details: '',
      label: '',
      idCounter: 0,
      appointments: this.props.appointments
    };
    this.handleDateChange = this.handleDateChange.bind(this);
    this.handleTimeChange = this.handleTimeChange.bind(this);
    this.createAppointment = this.createAppointment.bind(this);
    this.updateDetails = this.updateDetails.bind(this);
  }

  handleDateChange(date) {
    this.setState({
      date: date
    });
  }

  handleTimeChange(time) {
    this.setState({
      time: time
    });
  }

  updateDetails(details) {
    this.setState({
      details: details.target.value
    });
  }


  createAppointment() {
    let appointments = this.state.appointments;

    let iDate = this.state.date.format("DD.MM.YYYY");
    let iTime = this.state.time.format("HH:mm");

    let dateAndTime = moment(iDate + " " + iTime);

    let id = this.state.idCounter;

    appointments.push([
      this.state.date.format("DD.MM.YYYY") + " " +
      this.state.time.format("HH:mm")
      ,this.state.details,
      dateAndTime,
      id]);
    this.setState({
      appointments: appointments,
      idCounter: this.state.idCounter + 1
    });
    this.props.onCreateAppointment(this.state.appointments);
    this.props.closeNewAppointment();
  };

  render() {
    return (
      <div className={'appointment-container'}>
        <Modal show={this.props.showModal} onHide={this.props.closeNewAppointment}>
          <Modal.Header closeButton>
            <Modal.Title>Modal heading</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <DatePicker
              selected={this.state.date}
              onChange={this.handleDateChange}
              dateFormat={"DD.MM.YYYY"}
            />
            <TimePicker
              showSecond={false}
              defaultValue={this.state.time}
              onChange={this.handleTimeChange}
            />
            <FormControl
              type={'text'}
              placeholder={''}
              value={this.state.details}
              onChange={this.updateDetails}
            />
            <Button
              onClick={this.createAppointment}>
              Create schedule!
            </Button>
          </Modal.Body>
        </Modal>
      </div>
    )
  }
}

export default CreateAppointment;