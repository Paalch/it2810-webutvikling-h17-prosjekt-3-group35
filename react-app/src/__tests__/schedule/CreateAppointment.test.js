import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import CreateAppointment from "../../containers/schedule/CreateAppointment";


Enzyme.configure({ adapter: new Adapter() })


describe('Checks that the createNewAppointment component renders correctly', () => {

    it('Checks that newAppointment renders', () => {
        const rendered = renderer.create(
            <CreateAppointment
                onCreateAppointment={ () => {return false}}
                idCounter={0}
                showModal={false}
                closeNewAppointment={ () => {return false}}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });



});
