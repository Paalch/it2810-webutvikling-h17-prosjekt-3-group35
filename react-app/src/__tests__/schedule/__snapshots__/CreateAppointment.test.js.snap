// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the createNewAppointment component renders correctly Checks that newAppointment renders 1`] = `<div />`;
